solutions/usaco-1179.mdx

```diff
@@ -0,0 +1,90 @@
+---
+id: usaco-1179
+source: USACO Gold 2022 January
+title: Herdle
+author: Juheon Rhee
+---
+
+[Official Editorial (Java)](http://www.usaco.org/current/data/sol_prob1_bronze_jan22.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main(){
+
+	char correct[9], guess[9];
```

Comment:
please don't use arrays i'm begging you

Reply:
Ill switch to vector but why?

Reply:
hard to debug & use in general

Reply:
bump

Reply:
bump again

Reply:
bump for sans 😎

Reply:
can i plsjust use arrays i think in this case its easier to read

Reply:
> can i plsjust use arrays i think in this case its easier to read

you are physically not allowed to
refer to code conventions

Reply:
where does it say that you can't use arrays in the code conventions? i think arrays are fine...

Reply:
<img width="753" alt="Screen Shot 2022-06-18 at 4 42 45 PM" src="https://user-images.githubusercontent.com/79723780/174456489-90a6e220-0485-4e8d-aba0-a5f54fbc8952.png">

Reply:
those aren't variable length arrays...

Reply:
they arent that far off...

Reply:
umm the reason for disallowing variable length arrays is that they don't work with all compilers and so are not valid c++, while the statements above are valid

Reply:
could you not use a string? wouldn't that be simpler?

Reply:
there's nothing wrong with using an array... it might be a little bit better to use vectors, but not sure it really matters